feature/bgp/policybase/otg_tests/route_installation_test/route_installation_test.go

```diff
@@ -0,0 +1,683 @@
+		t.Errorf("BGP peerAs: got %v, want %v", peerAS, ateAS)
+	}
+
+	// Check BGP neighbor is enabled
+	if !nbrPath.Get(t).GetEnabled() {
+		t.Errorf("Expected neighbor %v to be enabled", ateSrc.IPv4)
+	}
+}
+
+// verifyPrefixesTelemetry confirms that the dut shows the correct numbers of installed, sent and
+// received IPv4 prefixes
+// TODO: Need to refactor and compare using cmp.diff
+func verifyPrefixesTelemetry(t *testing.T, dut *ondatra.DUTDevice, wantInstalled, wantRx, wantSent uint32) {
+	statePath := dut.Telemetry().NetworkInstance("default").Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	prefixesv4 := statePath.Neighbor(ateDst.IPv4).AfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).Prefixes()
+	if gotInstalled := prefixesv4.Installed().Get(t); gotInstalled != wantInstalled {
+		t.Errorf("Installed prefixes mismatch: got %v, want %v", gotInstalled, wantInstalled)
+	}
+	if gotRx := prefixesv4.ReceivedPrePolicy().Get(t); gotRx != wantRx {
+		t.Errorf("Received prefixes mismatch: got %v, want %v", gotRx, wantRx)
+	}
+	if gotSent := prefixesv4.Sent().Get(t); gotSent != wantSent {
+		t.Errorf("Sent prefixes mismatch: got %v, want %v", gotSent, wantSent)
+	}
+}
+
+// verifyPrefixesTelemetryV6 confirms that the dut shows the correct numbers of installed, sent and
+// received IPv6 prefixes
+func verifyPrefixesTelemetryV6(t *testing.T, dut *ondatra.DUTDevice, wantInstalledv6, wantRxv6, wantSentv6 uint32) {
+	statePath := dut.Telemetry().NetworkInstance("default").Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	prefixesv6 := statePath.Neighbor(ateDst.IPv6).AfiSafi(telemetry.BgpTypes_AFI_SAFI_TYPE_IPV6_UNICAST).Prefixes()
+
+	if gotInstalledv6 := prefixesv6.Installed().Get(t); gotInstalledv6 != wantInstalledv6 {
+		t.Errorf("IPV6 Installed prefixes mismatch: got %v, want %v", gotInstalledv6, wantInstalledv6)
+	}
+	if gotRxv6 := prefixesv6.ReceivedPrePolicy().Get(t); gotRxv6 != wantRxv6 {
+		t.Errorf("IPV6 Received prefixes mismatch: got %v, want %v", gotRxv6, wantRxv6)
+	}
+	if gotSentv6 := prefixesv6.Sent().Get(t); gotSentv6 != wantSentv6 {
+		t.Errorf("IPv6 Sent prefixes mismatch: got %v, want %v", gotSentv6, wantSentv6)
+	}
+}
+
+// verifyPolicyTelemetry confirms that the dut policy is set as expected.
+func verifyPolicyTelemetry(t *testing.T, dut *ondatra.DUTDevice, policy string) {
+	statePath := dut.Telemetry().NetworkInstance("default").Protocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, "BGP").Bgp()
+	policytel := statePath.PeerGroup(peerGrpName).ApplyPolicy().ImportPolicy().Get(t)
+	for _, val := range policytel {
+		if val != policy {
+			t.Errorf("Apply policy mismatch: got %v, want %v", policytel, policy)
+		}
+	}
+}
+
+// configureOTG configures the interfaces and BGP protocols on an OTG, including advertising some
+// (faked) networks over BGP.
+func configureOTG(t *testing.T, otg *ondatra.OTG, expectedRoutes int32) (gosnappi.Config, otgutils.ExpectedState) {
```

Comment:
Looking at the pairwise diff between the ATE and OTG version, I can't help but notice that the OTG version is longer, partly because it wants a name assigned for every node in the config tree. Can some of these names be omitted?

Reply:
Unfortunately setting the name is mandatory in the model for every object/node. Maybe the model could be improved by adding some deafult values to names, mac addresses (see the previous comments)...etc

Reply:
I just realized that telemetry will need the node names.

Since node names are mandatory, how about making it more streamlined?

```
config := otg.NewConfig(t)
srcDev := config.GetOrCreateDevice("src") // or just config.Device()
srcPort := srcDev.GetOrCreatePort("port1") // or just srcDev.Port()
srcEth := srcPort.GetOrCreateEthernet("eth1").SetMAC(...)
srcIPv4 := srcEth.GetOrCreateIPv4("eth1:ipv4").SetAddress(...).SetGateway(...)
```

Reply:
I made some changes...see below, but for a function like "GetOrCreateDevice" we would need to create a new wrapper as gosnappi doesn't support it.
```go
	srcDev := config.Devices().Add().SetName(ateSrc.Name)
	srcEth := srcDev.Ethernets().Add().SetName(ateSrc.Name + ".Eth")
	srcEth.SetPortName(srcPort.Name()).SetMac(ateSrc.MAC)
	srcIpv4 := srcEth.Ipv4Addresses().Add().SetName(ateSrc.Name + ".IPv4")
	srcIpv4.SetAddress(ateSrc.IPv4).SetGateway(dutSrc.IPv4).SetPrefix(int32(ateSrc.IPv4Len))
```